### PR TITLE
soc: atmel: Cleanup dts_fixup.h

### PR DESCRIPTION
**Pull request by galak**
Remove dts_fixup.h files that are needed anymore, remove defines that
are used, and replace defines with new DT macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: soc/arm/atmel_sam/same70/dts_fixup.h [hunk 1]
```diff
@@ -8,22 +8,7 @@
...
-#define DT_SPI_1_IRQ_PRI		DT_ATMEL_SAM_SPI_40058000_IRQ_0_PRIORITY
-#define DT_SPI_1_PERIPHERAL_ID	DT_ATMEL_SAM_SPI_40058000_PERIPHERAL_ID
-
-#define DT_ADC_0_NAME	DT_ATMEL_SAM_AFEC_4003C000_LABEL
-#define DT_ADC_1_NAME	DT_ATMEL_SAM_AFEC_40064000_LABEL
-
-#define DT_FLASH_DEV_BASE_ADDRESS		DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_BASE_ADDRESS
-#define DT_FLASH_DEV_NAME			DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_LABEL
+#define DT_ADC_0_NAME	DT_LABEL(DT_NODELABEL(afec0))
+#define DT_ADC_1_NAME	DT_LABEL(DT_NODELABEL(afec0))
```
Comment (nandojve) [inline soc/arm/atmel_sam/same70/dts_fixup.h:12]: afec1 maybe?

File: soc/arm/atmel_sam/samv71/dts_fixup.h [hunk 2]
```diff
@@ -12,22 +12,7 @@
...
-#define DT_SPI_1_IRQ_PRI                     DT_ATMEL_SAM_SPI_40058000_IRQ_0_PRIORITY
-#define DT_SPI_1_PERIPHERAL_ID               DT_ATMEL_SAM_SPI_40058000_PERIPHERAL_ID
-
-#define DT_ADC_0_NAME                        DT_ATMEL_SAM_AFEC_4003C000_LABEL
-#define DT_ADC_1_NAME                        DT_ATMEL_SAM_AFEC_40064000_LABEL
-
-#define DT_FLASH_DEV_BASE_ADDRESS            DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_BASE_ADDRESS
-#define DT_FLASH_DEV_NAME                    DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_LABEL
+#define DT_ADC_0_NAME	DT_LABEL(DT_NODELABEL(afec0))
+#define DT_ADC_1_NAME	DT_LABEL(DT_NODELABEL(afec0))
```
Comment (nandojve) [inline soc/arm/atmel_sam/samv71/dts_fixup.h:16]: same here..

**Review by nandojve — CHANGES_REQUESTED** [review]

File: soc/arm/atmel_sam/same70/dts_fixup.h (hunk 1 above)
Comment (galak) [reply soc/arm/atmel_sam/same70/dts_fixup.h]: fixed

File: soc/arm/atmel_sam/samv71/dts_fixup.h (hunk 2 above)
Comment (galak) [reply soc/arm/atmel_sam/samv71/dts_fixup.h]: fixed

**Review by nandojve — APPROVED** [review]

File: soc/arm/atmel_sam/same70/dts_fixup.h [hunk 3]
```diff
@@ -8,22 +8,7 @@
...
-#define DT_SPI_1_IRQ_PRI		DT_ATMEL_SAM_SPI_40058000_IRQ_0_PRIORITY
-#define DT_SPI_1_PERIPHERAL_ID	DT_ATMEL_SAM_SPI_40058000_PERIPHERAL_ID
-
-#define DT_ADC_0_NAME	DT_ATMEL_SAM_AFEC_4003C000_LABEL
-#define DT_ADC_1_NAME	DT_ATMEL_SAM_AFEC_40064000_LABEL
-
-#define DT_FLASH_DEV_BASE_ADDRESS		DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_BASE_ADDRESS
-#define DT_FLASH_DEV_NAME			DT_ATMEL_SAM_FLASH_CONTROLLER_400E0C00_LABEL
+#define DT_ADC_0_NAME	DT_LABEL(DT_NODELABEL(afec0))
+#define DT_ADC_1_NAME	DT_LABEL(DT_NODELABEL(afec1))
```
Comment (stephanosio) [inline soc/arm/atmel_sam/same70/dts_fixup.h:12]: @galak Is there an upcoming fix to get rid of these? (e.g. converting these to `chosen` or something like that)

Comment (galak) [reply soc/arm/atmel_sam/same70/dts_fixup.h:12]: I don't have a plan yet.  `chosen` doesn't really solve the issue of how these are used by the shell.

**Pull request merged**